app/services/users_service.rb

```diff
@@ -19,7 +20,7 @@ def register(email, password, organization_name)
     result.user = User.find_or_initialize_by(email: email)
 
     if result.user.id
-      result.fail!(code: 'user_already_exists')
+      result.single_validation_failure!(field: :email, error_code: 'user_already_exists')
```

Comment:
Maybe changing from `user_already_exists` to `already_exists` as it's now linked to the `email` field?

Reply:
Good one indeed. As it impacts the front-end, I will handle it in another PR